doc/cephfs/charmap.rst

```diff
@@ -0,0 +1,185 @@
+CephFS Directory Entry Name Normalization and Case Folding
+==========================================================
+
+CephFS allows configuring directory trees to **normalize** or **case fold**
+directory entry names. This is typically a useful property for file systems
+exported by gateways like Samba which enforce a case-insensitive view of the
+file system, typically with performance penalties on file systems which are not
+case-insensitive.
+
+The following virtual extended attributes control the **character mapping**
+rules for directory entries:
+
+* ``ceph.dir.casesensitive``: A boolean setting for the case sensitivity of the directory.
+* ``ceph.dir.normalization``: A string setting for the type of Unicode normalization to use. Currently the normalization forms D (``nfd``), C (``nfc``), KD (``nfkd``), and KC (``nfkc``) are understood by the client.
+* ``ceph.dir.encoding``: A string setting for the encoding to use and enforce for directory entry names. The default and presently only supported encoding is UTF-8 (``utf8``).
+
+There is also a convenience virtual extended attribute that is useful for
+getting the JSON encoding of the case sensitivity, normalization, and encoding
+configurations:
+
+* ``ceph.dir.charmap``: The complete character mapping configuration for a directory.
+
+It can also be used to **remove** all settings and restore the default CephFS behavior
+for directory entry names (uninterpreted bytes without ``/`` that are NUL terminated).
+
+Note the following restriction on manipulating any of these extended attributes:
+
+* The directory must be empty.
+* The directory must not be part of a snapshot.
+
+New sub-directories created under a directory with a **charmap** configuration will
```

Comment:
Suggest double backticks instead of double asterisks.

Also, it would seem that both `subdirectories` and `sub-directories` are accepted.  I find ~35 instances of the former and ~16 of the latter in the docs, suggest that we converged on `subdirectories`.

Reply:
done in next push